ui/src/components/App.js

```diff
@@ -21,65 +18,105 @@ import HelpContainer from '../containers/HelpContainer';
 import Main from './Main';
 import LoadingScreen from '../components/LoadingScreen/LoadingScreen';
 
-async function requireAuth() {
-  await store.dispatch(getLoginInfo());
-  const { login } = store.getState();
+import { store } from '../store';
+import { serverIO } from '../serverIO';
 
-  if (login.loggedIn) {
-    await store.dispatch(startSession(login.user.inControl));
-  }
+import { getLoginInfo } from '../actions/login';
+import { getInitialState, applicationFetched } from '../actions/general';
 
-  if (login.loggedIn && !serverIO.initialized) {
-    serverIO.listen(store);
+export async function requireAuth() {
+  try {
+    store.dispatch(applicationFetched(false));
+    await store.dispatch(getLoginInfo());
+  } catch {
+    store.dispatch(applicationFetched(true));
+    return;
   }
+
+  await store.dispatch(getInitialState());
+  serverIO.listen(store);
```

Comment:
Need to disconnect in the `useEffect`'s clean-up function to avoid memory leaks: https://react.dev/reference/react/useEffect#parameters